### ISSUE TITLE
[FR] Add support for Wake Words in Companion app/on Android devices

### ISSUE THREAD

**Issue by AleXSR700** [home-assistant/android#4201]
**Is your feature request related to a problem? Please describe.**

When an Android phone or tablet  is not reachable but is within hearing range, text commands cannot be sent to the assistant, but using Wake Words, it would be possible to send commands even without physicially accessing the devices.

**Describe the solution you'd like**

Implement support for wake words, e.g. using OpenWakeWords.

**Describe alternatives you've considered, if any**

None available

**Additional context**

**Comment by HarryBak** [comment]
Anyone got any insights on limitations here? I'm extremely interested in helping move this forward.

**Comment by borgqueenx** [comment]
Hope this can be added soon...would be excellent to control our home with voice using the companion app.

**Comment by Hadatko** [comment]
+1 My device is waiting for this feature. Meantime i use atom echo device.

**Comment by AleXSR700** [comment]
Any information if this will make it into the app, say, in the next 6 months?

**Comment by jaecktec** [comment]
It's an open source project - if it's urgent you can propose a PR :)

**Comment by AleXSR700** [comment]
> It's an open source project - if it's urgent you can propose a PR :) 
> 
> I found this as a technical demonstration: 
> https://github.com/tuckercr/wakewordapp

If I could, I would ;-)

Unfortunately, in this case, I am not a programmer :-(

**Comment by AleXSR700** [comment]
Friendly push :)

**Comment by kushalgoaten** [comment]
how do  i vote for this?

**Comment by SmoochSmackers** [comment]
IDK exactly what qualifies as voting for this, but this would be amazing! 

Never could get the termux method working on my phone, so currently I am having to use hassmic to constantly stream my phone's audio to my HA server, which isn't the best for battery life and is wildly inefficient, but credit to hassmic since it does work.

**Comment by pilot51** [comment]
@kushalgoaten @SmoochSmackers Vote for it with the 👍 reaction.

**Comment by AleXSR700** [comment]
> @kushalgoaten @SmoochSmackers Vote for it with the 👍 reaction.

In the first post, for those confused ;)

**Comment by rsmanning** [comment]
It seems like a no-brainer to and this considering there big push for AI functionality in-home assistant. The only thing I can think of is they're trying to sell more of their voice assistant boxes.

**Comment by tron1point0** [comment]
Those are static base stations; I'm not taking one with me to the store.

**Comment by TimoPtr** [comment]
> It seems like a no-brainer to and this considering there big push for AI functionality in-home assistant. The only thing I can think of is they're trying to sell more of their voice assistant boxes.

The majority of the contributions to this application come from volunteers who work on it in their free time. Anyone is welcome to develop and add this feature. There are many ideas and potential improvements for the app, but only a limited number of people who have the time or skills to implement them.

For anyone that wants to dig a bit more another user found some interesting classes from the Android documentation
https://github.com/home-assistant/android/discussions/5974

**Comment by TimoPtr** [comment]
I've created a proof-of-concept using local microWakeWord detection (#6312). It still needs significant work before release.

  Unfortunately, wake word detection requires continuous listening. Google has restricted the `HotwordDetection` API behind `@SystemApi`, preventing third-party apps from using the DSP for low-power wake word detection (like "Ok Google" does). As a result, we must keep the CPU active while listening, which will impact battery life.

**Comment by winmutt** [comment]
> I've created a proof-of-concept using local microWakeWord detection ([#6312](https://github.com/home-assistant/android/pull/6312)). It still needs significant work before release.
> 
> Unfortunately, wake word detection requires continuous listening. Google has restricted the `HotwordDetection` API behind `@SystemApi`, preventing third-party apps from using the DSP for low-power wake word detection (like "Ok Google" does). As a result, we must keep the CPU active while listening, which will impact battery life.

Thanks! Excited to try this out. I suspect a lot of use cases will be for fixed base android devices rather than mobile ones. That is very disappointing about the Google API.

**Comment by signaleleven** [comment]
If #6312 is the feature currently available in the companion app, it works. It's a shame about the API gating, and I would assume that it breaks some antimonopoly law :)

A necessary improvement would be a chime to notify that the trigger happened (like in other Assist Devices). If the screen is off, or if you cannot look at the phone, there is no indication that the wake word was detected.

**Comment by pilot51** [comment]
> A necessary improvement would be a chime to notify that the trigger happened (like in other Assist Devices). If the screen is off, or if you cannot look at the phone, there is no indication that the wake word was detected.

Agreed. I often use it with my phone in my pocket and it's very hit-and-miss whether it picks up the hotword, so I say the full command and wait a few seconds for nothing to happen and try again. While a chime wouldn't make the hotword more reliable, it would give me the assurance that I'm not wasting my time and breath on a command it isn't listening to.

**Comment by Gronsak** [comment]
Would the latest EU decision mean we would get access to the `HotwordDetection` api or am I reading it wrong? 

https://digital-markets-act.ec.europa.eu/developer-portal/interoperability/alphabet-specification-proceedings-interoperability-ai-services_en

I'm guessing they're gonna drag their feet as long as possible on this though.

**Comment by jpelgrom** [comment]
> Would the latest EU decision mean we would get access to the `HotwordDetection` api or am I reading it wrong?

Yes, most likely, also see the blog post: https://www.openhomefoundation.org/blog/a-big-win-for-android-interoperability/
